<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1187</width>
    <height>844</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Методы отрисовки эллипсов</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QPushButton" name="pushButton_color_line">
    <property name="geometry">
     <rect>
      <x>790</x>
      <y>570</y>
      <width>111</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_color_fone">
    <property name="geometry">
     <rect>
      <x>790</x>
      <y>610</y>
      <width>111</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>630</x>
      <y>590</y>
      <width>171</width>
      <height>71</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Цвет фона:</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>630</x>
      <y>550</y>
      <width>201</width>
      <height>71</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Цвет рисования:</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_2">
    <property name="geometry">
     <rect>
      <x>630</x>
      <y>660</y>
      <width>171</width>
      <height>81</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Очистить</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_5">
    <property name="geometry">
     <rect>
      <x>980</x>
      <y>660</y>
      <width>191</width>
      <height>81</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="text">
     <string>график
окружностей</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_6">
    <property name="geometry">
     <rect>
      <x>910</x>
      <y>570</y>
      <width>131</width>
      <height>81</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>11</pointsize>
     </font>
    </property>
    <property name="text">
     <string>установить
фоновый</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_10">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>640</y>
      <width>51</width>
      <height>51</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="text">
     <string>0;0</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_11">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>0</y>
      <width>91</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="text">
     <string>0;600</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_12">
    <property name="geometry">
     <rect>
      <x>580</x>
      <y>0</y>
      <width>91</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="text">
     <string>600;600</string>
    </property>
   </widget>
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>650</x>
      <y>30</y>
      <width>481</width>
      <height>291</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="tab">
     <attribute name="title">
      <string>Окружность</string>
     </attribute>
     <widget class="QLineEdit" name="lineEdit_x_c">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>80</y>
        <width>81</width>
        <height>41</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>16</pointsize>
       </font>
      </property>
      <property name="text">
       <string>300</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_13">
      <property name="geometry">
       <rect>
        <x>110</x>
        <y>10</y>
        <width>281</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>14</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Одиночная окружность:</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_14">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>50</y>
        <width>101</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>14</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Центр X</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="lineEdit_y_c">
      <property name="geometry">
       <rect>
        <x>130</x>
        <y>80</y>
        <width>81</width>
        <height>41</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>16</pointsize>
       </font>
      </property>
      <property name="text">
       <string>300</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_15">
      <property name="geometry">
       <rect>
        <x>130</x>
        <y>50</y>
        <width>91</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>14</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Центр Y</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="lineEdit_r_c">
      <property name="geometry">
       <rect>
        <x>230</x>
        <y>80</y>
        <width>81</width>
        <height>41</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>16</pointsize>
       </font>
      </property>
      <property name="text">
       <string>100</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_16">
      <property name="geometry">
       <rect>
        <x>240</x>
        <y>50</y>
        <width>111</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>14</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Радиус</string>
      </property>
     </widget>
     <widget class="QPushButton" name="one_circle">
      <property name="geometry">
       <rect>
        <x>330</x>
        <y>80</y>
        <width>91</width>
        <height>41</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>14</pointsize>
       </font>
      </property>
      <property name="text">
       <string>OK</string>
      </property>
      <property name="shortcut">
       <string>Return</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_17">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>140</y>
        <width>321</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>14</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Концентрические окружности:</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="lineEdit_r_k_c">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>200</y>
        <width>81</width>
        <height>41</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>16</pointsize>
       </font>
      </property>
      <property name="text">
       <string>100</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_20">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>170</y>
        <width>81</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>14</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Радиус</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_k_c">
      <property name="geometry">
       <rect>
        <x>330</x>
        <y>200</y>
        <width>91</width>
        <height>41</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>14</pointsize>
       </font>
      </property>
      <property name="text">
       <string>OK</string>
      </property>
      <property name="shortcut">
       <string>Return</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="lineEdit_dr_k_c">
      <property name="geometry">
       <rect>
        <x>130</x>
        <y>200</y>
        <width>81</width>
        <height>41</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>16</pointsize>
       </font>
      </property>
      <property name="text">
       <string>10</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="lineEdit_n_k_c">
      <property name="geometry">
       <rect>
        <x>230</x>
        <y>200</y>
        <width>81</width>
        <height>41</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>16</pointsize>
       </font>
      </property>
      <property name="text">
       <string>3</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_21">
      <property name="geometry">
       <rect>
        <x>150</x>
        <y>170</y>
        <width>81</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>14</pointsize>
       </font>
      </property>
      <property name="text">
       <string>d(R)</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_22">
      <property name="geometry">
       <rect>
        <x>260</x>
        <y>170</y>
        <width>51</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>14</pointsize>
       </font>
      </property>
      <property name="text">
       <string>N</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_2">
     <attribute name="title">
      <string>Эллипс</string>
     </attribute>
     <widget class="QLineEdit" name="lineEdit_a_e">
      <property name="geometry">
       <rect>
        <x>190</x>
        <y>80</y>
        <width>81</width>
        <height>41</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>16</pointsize>
       </font>
      </property>
      <property name="text">
       <string>150</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_19">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>50</y>
        <width>101</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>14</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Центр X</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_26">
      <property name="geometry">
       <rect>
        <x>200</x>
        <y>170</y>
        <width>81</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>14</pointsize>
       </font>
      </property>
      <property name="text">
       <string>delta</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="lineEdit_x_e">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>80</y>
        <width>81</width>
        <height>41</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>16</pointsize>
       </font>
      </property>
      <property name="text">
       <string>300</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_28">
      <property name="geometry">
       <rect>
        <x>310</x>
        <y>170</y>
        <width>51</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>14</pointsize>
       </font>
      </property>
      <property name="text">
       <string>N</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_18">
      <property name="geometry">
       <rect>
        <x>110</x>
        <y>50</y>
        <width>91</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>14</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Центр Y</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_k_e">
      <property name="geometry">
       <rect>
        <x>370</x>
        <y>200</y>
        <width>91</width>
        <height>41</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>14</pointsize>
       </font>
      </property>
      <property name="text">
       <string>OK</string>
      </property>
      <property name="shortcut">
       <string>Return</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_23">
      <property name="geometry">
       <rect>
        <x>240</x>
        <y>50</y>
        <width>41</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>14</pointsize>
       </font>
      </property>
      <property name="text">
       <string>a</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="lineEdit_del_k_e">
      <property name="geometry">
       <rect>
        <x>190</x>
        <y>200</y>
        <width>81</width>
        <height>41</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>16</pointsize>
       </font>
      </property>
      <property name="text">
       <string>20</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="lineEdit_y_e">
      <property name="geometry">
       <rect>
        <x>100</x>
        <y>80</y>
        <width>81</width>
        <height>41</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>16</pointsize>
       </font>
      </property>
      <property name="text">
       <string>300</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_24">
      <property name="geometry">
       <rect>
        <x>90</x>
        <y>10</y>
        <width>281</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>14</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Одиночный эллипс:</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_25">
      <property name="geometry">
       <rect>
        <x>50</x>
        <y>140</y>
        <width>341</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>14</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Концентрические эллипсы:</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_9">
      <property name="geometry">
       <rect>
        <x>370</x>
        <y>80</y>
        <width>91</width>
        <height>41</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>14</pointsize>
       </font>
      </property>
      <property name="text">
       <string>OK</string>
      </property>
      <property name="shortcut">
       <string>Return</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="lineEdit_n_k_e">
      <property name="geometry">
       <rect>
        <x>280</x>
        <y>200</y>
        <width>81</width>
        <height>41</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>16</pointsize>
       </font>
      </property>
      <property name="text">
       <string>5</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="lineEdit_b_e">
      <property name="geometry">
       <rect>
        <x>280</x>
        <y>80</y>
        <width>81</width>
        <height>41</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>16</pointsize>
       </font>
      </property>
      <property name="text">
       <string>50</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_29">
      <property name="geometry">
       <rect>
        <x>330</x>
        <y>50</y>
        <width>41</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>14</pointsize>
       </font>
      </property>
      <property name="text">
       <string>b</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="lineEdit_a_k_e">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>200</y>
        <width>81</width>
        <height>41</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>16</pointsize>
       </font>
      </property>
      <property name="text">
       <string>150</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_30">
      <property name="geometry">
       <rect>
        <x>130</x>
        <y>170</y>
        <width>41</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>14</pointsize>
       </font>
      </property>
      <property name="text">
       <string>b</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="lineEdit_b_k_e">
      <property name="geometry">
       <rect>
        <x>100</x>
        <y>200</y>
        <width>81</width>
        <height>41</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>16</pointsize>
       </font>
      </property>
      <property name="text">
       <string>50</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_27">
      <property name="geometry">
       <rect>
        <x>40</x>
        <y>170</y>
        <width>41</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>14</pointsize>
       </font>
      </property>
      <property name="text">
       <string>a</string>
      </property>
     </widget>
    </widget>
   </widget>
   <widget class="QWidget" name="layoutWidget_2">
    <property name="geometry">
     <rect>
      <x>650</x>
      <y>320</y>
      <width>421</width>
      <height>211</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_3">
     <item>
      <widget class="QRadioButton" name="radioButton_canonich">
       <property name="font">
        <font>
         <pointsize>14</pointsize>
        </font>
       </property>
       <property name="text">
        <string>Каноническое уравнение</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="radioButton_param">
       <property name="font">
        <font>
         <pointsize>14</pointsize>
        </font>
       </property>
       <property name="text">
        <string>Параметрическое уравнение</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="radioButton_br">
       <property name="font">
        <font>
         <pointsize>14</pointsize>
        </font>
       </property>
       <property name="text">
        <string>Брезенхем</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="radioButton_midle">
       <property name="font">
        <font>
         <pointsize>14</pointsize>
        </font>
       </property>
       <property name="text">
        <string>Алгоритм средней точки</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="radioButton_standart">
       <property name="font">
        <font>
         <pointsize>14</pointsize>
        </font>
       </property>
       <property name="text">
        <string>Стандартный</string>
       </property>
       <property name="checked">
        <bool>false</bool>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="GLWidget" name="widget" native="true">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>40</y>
      <width>600</width>
      <height>600</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_7">
    <property name="geometry">
     <rect>
      <x>1040</x>
      <y>570</y>
      <width>131</width>
      <height>81</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>11</pointsize>
     </font>
    </property>
    <property name="text">
     <string>вернуть
предыдущий</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_8">
    <property name="geometry">
     <rect>
      <x>800</x>
      <y>660</y>
      <width>181</width>
      <height>81</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="text">
     <string>график
эллипсов</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1187</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>GLWidget</class>
   <extends>QWidget</extends>
   <header>glwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>lineEdit_x_c</tabstop>
  <tabstop>lineEdit_y_c</tabstop>
  <tabstop>lineEdit_r_c</tabstop>
  <tabstop>one_circle</tabstop>
  <tabstop>lineEdit_r_k_c</tabstop>
  <tabstop>lineEdit_dr_k_c</tabstop>
  <tabstop>lineEdit_n_k_c</tabstop>
  <tabstop>pushButton_k_c</tabstop>
  <tabstop>radioButton_canonich</tabstop>
  <tabstop>radioButton_param</tabstop>
  <tabstop>radioButton_br</tabstop>
  <tabstop>radioButton_midle</tabstop>
  <tabstop>radioButton_standart</tabstop>
  <tabstop>pushButton_6</tabstop>
  <tabstop>lineEdit_x_e</tabstop>
  <tabstop>lineEdit_y_e</tabstop>
  <tabstop>lineEdit_a_e</tabstop>
  <tabstop>lineEdit_b_e</tabstop>
  <tabstop>pushButton_9</tabstop>
  <tabstop>lineEdit_a_k_e</tabstop>
  <tabstop>lineEdit_b_k_e</tabstop>
  <tabstop>lineEdit_del_k_e</tabstop>
  <tabstop>lineEdit_n_k_e</tabstop>
  <tabstop>pushButton_k_e</tabstop>
  <tabstop>pushButton_5</tabstop>
  <tabstop>tabWidget</tabstop>
  <tabstop>pushButton_color_line</tabstop>
  <tabstop>pushButton_color_fone</tabstop>
  <tabstop>pushButton_2</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
